Sources/SotoCore/Credential/ConfigFileCredentialProvider.swift

```diff
@@ -133,6 +133,14 @@ struct AWSConfigFileCredentialProvider: CredentialProvider {
 
             return pth
         }
+        #elseif os(macOS)
+        // can not use wordexp on macOS because for sandboxed application wexp.we_wordv == nil
+        guard let home = getpwuid(getuid())?.pointee.pw_dir,
+            let homePath = String(cString: home, encoding: .utf8)
+        else {
+            return filePath
+        }
+        return filePath.replacingOccurrences(of: "~", with: homePath)
```

Comment:
Should this only be replaced if it is the first character of the path?

Reply:
although it is very unlikely to have other tildes in such path, you're right !

Reply:
It's better we are correct. Look up bash tilda expansion and it gets all complex. Let's go for checking as a prefix

Reply:
yep, I just changed it